phoenix-core/src/main/java/org/apache/phoenix/compile/StatementPlan.java

```diff
@@ -29,4 +33,6 @@
     ParameterMetaData getParameterMetaData();
     
     ExplainPlan getExplainPlan() throws SQLException;
+    public Set<TableRef> getSourceRefs();
```

Comment:
This interface change (along with the one in MutationPlan) caused some diffs not related to transactions. If it's easy to merge that back separately, that might help reduce the extraneous, non transactional changes.